src/tests/poc/new-workflow/simple-test.cpp

```diff
@@ -0,0 +1,119 @@
+#define BOOST_TEST_MODULE poc tests
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/data/test_case.hpp>
+
+namespace tt = boost::test_tools;
+
+#include "antares/optim/api/LinearProblemBuilder.h"
+#include "antares/optim/impl/LinearProblemImpl.h"
+#include "../include/simple/Battery.h"
+#include "../include/simple/Thermal.h"
+#include "../include/simple/Balance.h"
+#include "../include/simple/ProductionPriceMinimization.h"
+
+using namespace Antares::optim::api;
+
+BOOST_AUTO_TEST_CASE(test_oneWeek_oneNode_oneBattery_oneThermal)
+{
+    vector<int> timeStamps{0, 1, 2, 3}; // toujours commencer à 0 sinon ça plante actuellement
+    int timeResolution = 60;
+
+    LinearProblemImpl linearProblem(false, "xpress");
+    LinearProblemBuilder linearProblemBuilder(linearProblem);
+
+    Battery battery("battery1", 100, 1000);
+    Thermal thermal("thermal1", 100);
+    Balance balance("nodeA", {&battery}, {&thermal});
+    ProductionPriceMinimization objective({&thermal});
+
+    linearProblemBuilder.addFiller(battery);
+    linearProblemBuilder.addFiller(thermal);
+    linearProblemBuilder.addFiller(balance);
+    linearProblemBuilder.addFiller(objective);
+
+    LinearProblemData linearProblemData(
+            timeStamps, // TODO : move to LinearProblem ?
+            timeResolution, // TODO : move to LinearProblem ?
+            {
+                    {"initialStock_battery1", 0}
+            },
+            {
+                    {"consumption_nodeA", {50, 50, 150, 120}},
+                    {"cost_thermal1",     {1,  3,  10,  8}}
+            });
+    linearProblemBuilder.build(linearProblemData);
+    auto solution = linearProblemBuilder.solve();
+
+    // la conso est supérieure à la Pmax du thermique sur les pas de temps 2 & 3
+    // Donc la batterie doit se charger pendant les TS 0 & 1
+    // En plus, le coût de la production est supérieur en 2 & 3, donc la batterie devrait se charger à fond en 0 & 1
+    // La conso en 0 & 1 est de 50, donc la batterie peut se recharger de 50
+    // Elle répartira sa charge surtout en 2 (prix thermique plus cher qu'en 3), en gardant au moins 20
+    // pour le pdt 3, dans lequel la conso est de 120 et la prod thermique max de 100
+    // Le thermique complète
+    // d'où le plan optimal
+    // thermique : 100, 100, 70, 100
+    // batterie : -50, -50, 80, 20
+
+    vector<double> actualThermalP = solution.getOptimalValues({"P_thermal1_0", "P_thermal1_1", "P_thermal1_2", "P_thermal1_3"});
+    vector<double> expectedThermalP({100., 100., 70., 100.});
+    BOOST_TEST(actualThermalP == expectedThermalP, tt::per_element()); // TODO add tolerance?
```

Comment:
yes add tolerance

Reply:
will do but boost does not allow it out of the box